mc-mark-more.el

```diff
@@ -30,17 +30,33 @@
 (require 'multiple-cursors-core)
 (require 'thingatpt)
 
+(defcustom mc/more-like-this-no-match-behavior 'error
+  "What to do if asked to match beyond the last cursor or before the first cursor."
+  :type '(radio (const :tag "Loop around to beginning/end of document." continue)
+                (const :tag "Warn and then loop around." warn)
+                (const :tag "Signal an error." error))
+  :group 'multiple-cursors)
+(setq mc/more-like-this-no-match-behavior 'continue)
+
+(defun mc/handle-more-like-this-no-match-condition (error-message)
+  (cl-ecase mc/more-like-this-no-match-behavior
+    (error (error error-message))
+    (warn  (message error-message))
+    (continue 'continue)))
+
 (defun mc/cursor-end (cursor)
-  (if (overlay-get cursor 'mark-active)
-      (max (overlay-get cursor 'point)
-           (overlay-get cursor 'mark))
-    (overlay-get cursor 'point)))
+  (when cursor
```

Comment:
Without that:
```
(end-char (cl-ecase direction
            (forwards  (or (mc/cursor-end (mc/nearest-cursor-before-point)) (point-min)))
            (backwards (or (mc/cursor-beg (mc/nearest-cursor-after-point)) (point-max)))))
```
Could fail because `(mc/nearest-cursor-before-point)` could be `nil`. Which would lead to:
```
(end-char (cl-ecase direction
            (forwards  (or (and (mc/nearest-cursor-before-point) (mc/cursor-end (mc/nearest-cursor-before-point))) (point-min)))
            (backwards (or (and (mc/nearest-cursor-before-point) (mc/cursor-beg (mc/nearest-cursor-after-point))) (point-max)))))
```
It felt cleaner to refactor by modifying `mc/cursor-beg|end`.

Reply:
But now those can return nil so the caller can't really rely on that anyway.

You can use `-if-let` from dash:

```
(-if-let (cursor (nearest-cursor)) (mc/begin cursor) (point-min))
```

Reply:
Ahhh I didn't know about `-if-let`, neat! `mc` doesn't really on `dash` yet and I feel like it is a bit much to add a dependency just for this. I've opted to using the built-in `if-let*`

Reply:
Ahhhh not built-in on older versions of emacs :/

I'll have a look at making this neater in a bit!